src/storage/hub.ts

```diff
@@ -40,16 +41,24 @@ export async function uploadToGaiaHub(
   filename: string, 
   contents: Blob | Buffer | ArrayBufferView | string,
   hubConfig: GaiaHubConfig,
-  contentType: string = 'application/octet-stream'
-): Promise<string> {
+  contentType: string = 'application/octet-stream',
+  etag?: string
+): Promise<PutFileResponse> {
   Logger.debug(`uploadToGaiaHub: uploading ${filename} to ${hubConfig.server}`)
+
+  const headers: { [key: string]: string; } = {
+    'Content-Type': contentType,
+    Authorization: `bearer ${hubConfig.token}`
+  }
+
+  if (etag) {
+    headers['If-Match'] = etag
```

Comment:
It seems like this is one of the few headers that we will have available in the future for _any_ typeof metadata transmission. 
Should we send structured payload for versioning/future-proofing reasons?
e.g.
```js
headers['If-Match'] = JSON.stringify({
  'v': '1.0',
  'tag': etag,
})
```

Reply:
Identified some potential issues where something like may be needed on initial release https://github.com/blockstack/gaia/pull/276/files#r358970449